src/docs/patterns/tables.mdx

```diff
@@ -0,0 +1,463 @@
+
+<MaterialDesignDescription title={'Data Table'} url={`https://material.io/components/data-tables`} />
+<TOC
+    anchors={[
+        { title: 'Introduction', hash: '#tables' },
+        { title: 'Appearance' },
+        { title: 'Behaviors' },
+        { title: 'Row Actions', depth: 1 },
+        { title: 'Batch Actions', depth: 1 },
+        { title: 'Sticky Header Row', depth: 1 },
+        { title: 'Expanded Row', depth: 1 },
+        { title: 'Horizontal Scrolling', depth: 1 },
+        { title: 'Searching and Filtering', depth: 1 },
+        { title: 'Transition Animation', depth: 1 },
+        { title: 'Tables vs. Lists' },
+        { title: 'Responsive' },
+        { title: 'Design Specifications' },
+        { title: 'Developers' },
+    ]}
+/>
+<Divider />
+
+# Appearance
+
+<ImageGrid images={[StandardTable]} caption={`A basic table.`} regularWidth />
+
+Table cells may be dynamic and interactive.
+
+<ImageGrid
+    images={[
+        TableCellText,
+        TableCellNumber,
+        TableCellTextField,
+        TableCellSwitch,
+        TableCellStatus,
+        TableCellIcon,
+        TableCellMultiSelect,
+        TableCellStripe,
+    ]}
+    captionsUnderImages={[
+        'Text content. The alternating background color is optional.',
+        'Numerical value. Note that both the header and the text cells are right-aligned.',
+        'Text fields. Note that text fields are right-aligned as they are expecting numerical values. These rows are also taller than standard table rows.',
+        'Switches and checkboxes.',
+        'Icons and tags.',
+        'Icon buttons at the end of the table row for touch screens.',
+        `Checkboxes to select rows.`,
+        `Status stripes for row items with special statuses.`,
+    ]}
+    gridImageProps={{ sm: 6, md: 4 }}
+/>
+
+You should seek to make your tables both easy to navigate and effective to use.
+
+<ImageGrid
+    images={[TableCellUnderlinedField, TableCellImage]}
+    captionsUnderImages={[
+        'CAUTION: Be careful when using underlined text fields in a table, as their understrokes compete with the line strokes between each table row.',
+        `AVOID: Images can heighten table rows. Consider using a list instead.`,
+    ]}
+/>
+
+# Behaviors
+
+## Row Actions
+
+Similar to lists, tables also have row actions. Common row actions include edit, delete and view details. They are usually placed at the end of a table row.
+
+<ImageGrid
+    images={[ActionAtRowEnd, ActionOnHover]}
+    captionsUnderImages={[
+        `Row actions can be placed at the end of the row.`,
+        `DO: On desktop, you can choose to have row actions visible only when the mouse is hovering on the row. The buttons won't compete for rooms with the rest of the table.`,
```

Comment:
```suggestion
        `DO: On desktop, you can choose to have row actions visible only when the mouse is hovering on the row. This will save space and make your tables cleaner and easier to read.`,
```